crates/router/src/db/merchant_account.rs

```diff
@@ -533,22 +543,62 @@ impl MerchantAccountInterface for MockDb {
         _publishable_key: &str,
     ) -> CustomResult<authentication::AuthenticationData, errors::StorageError> {
         // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let accounts = self.merchant_accounts.lock().await;
+        let account = accounts
+            .iter()
+            .find(|account| {
+                account
+                    .publishable_key
+                    .as_ref()
+                    .is_some_and(|key| key == _publishable_key)
```

Comment:
You can remove the underscore prefix from function params, wherever you're accessing them in your implementations.